core-bundle/src/Routing/RouteProvider.php

```diff
@@ -259,6 +259,7 @@ private function addRoutesForPage(PageModel $page, array &$routes): void
 
         $requirements = ['parameters' => '(/.+)?'];
         $path = sprintf('/%s{parameters}%s', $page->alias ?: $page->id, $this->urlSuffix);
+        $host = strtok((string) $page->domain, ':') ?: $page->domain;
```

Comment:
Since `$page->domain` can be empty, it is more effective to check this first. See

- https://3v4l.org/AZOrY/perf#output (current)
- https://3v4l.org/rDZm4/perf#output (new)

Reply:
AFAIK, `$page->domain` can be both `''` and `null`. That's why I am using `if($page->domain)` in 7741ba6375c098b929d20cefdbb3c5cd6d94c1c6.